pkg/services/publicdashboards/commands/generate_datasources/generate.go

```diff
@@ -0,0 +1,103 @@
+package generate_datasources
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"sync"
+)
+
+type listPluginResponse struct {
+	Items []struct {
+		Slug string `json:"slug"`
+	} `json:"items"`
+}
+
+type datasourceDetails struct {
+	Slug     string `json:"slug"`
+	Backend  bool   `json:"backend"`
+	Alerting bool   `json:"alerting"`
+}
+
+type datasource struct {
```

Comment:
We can do the same here

Reply:
`datasourceDetails` needs to be its own type bc its also used as the the return type for `getDatasourceDetails()`